src/components/Modal/ModalSelectAssets.tsx

```diff
@@ -80,43 +82,30 @@ export const ModalSelectAssets = () => {
       : undefined
 
     const getAssetList: SelectItemToken[] = []
-    const getAssetListWithBalances: SelectItemToken[] = []
+
     for (const [tokenId, token] of data) {
-      // We do not filter "tokenIn" as give full access to tokens in first step
-      // Filtration by routes should happen only at "tokenOut"
       const disabled = selectedTokenId != null && tokenId === selectedTokenId
-
-      if (isBaseToken(token)) {
-        if (token.balance != null) {
-          getAssetListWithBalances.push({
-            itemId: tokenId,
-            token,
-            disabled,
-            balance: {
-              balance: token.balance,
-              balanceUsd: token.balanceUsd,
-              convertedLast: token.convertedLast,
-            },
-          })
-        }
-      } else {
-        const hasBalance = token.groupedTokens.some(
-          (innerToken) => innerToken.balance != null
+      if (isUnifiedToken(token)) {
+        const defuseAssetId = token.groupedTokens.reduce(
```

Comment:
I didn't get why need this, even after reading the comment :) If it is needed for sure then you need to consider a better for it, because it is not `defuseAssetId`.

Reply:
For example, we need this on the deposit page, where we can't use the mt_balance call to retrieve a multi-asset grouped balance. Instead, we need to fetch balances by specific `defuseAssetIds`, such as USDC and ETH, rather than by `unifiedAssetId`.

Reply:
Won't be easier to pass an array instead of serializing to string and then deserializing?

Reply:
None of these options feel ideal to me. I’m inclined to leave it for now and revisit it when we have more time.